tests/auditd/test_auditd.py

```
@@ -0,0 +1,271 @@
+                         "/etc/cron.weekly",
+                         "/etc/cron.monthly"],
+        "dns_change": ["/etc/resolv.conf"],
+        "docker_config": ["/etc/docker/daemon.json"],
+        "docker_daemon": ["/usr/bin/dockerd"],
+        "docker_service": ["/lib/systemd/system/docker.service"],
+        "docker_socket": ["/lib/systemd/system/docker.socket"],
+        "modules_changes": ["/sbin/insmod",
+                            "/sbin/rmmod",
+                            "/sbin/modprobe"],
+        "log_changes": ["/var/log/testfile",
+                        "/var/log.tmpfs/testfile"],
+        "bin_changes": ["/bin/testfile"],
+        "sbin_changes": ["/sbin/testfile"],
+        "usr_bin_changes": ["/usr/bin/testfile"],
+        "usr_sbin_changes": ["/usr/sbin/testfile"],
+        "docker_storage": ["/var/lib/docker/testfile"]
+    }
+
+    for rule, files in watch_files.items():
+        for file in files:
+            ssh_remote_run(localhost,
+                           dutip,
+                           tacacs_creds['tacacs_rw_user'],
+                           tacacs_creds['tacacs_rw_user_passwd'],
+                           f"sudo touch {file}")
+            cmd = f"""show logging | grep 'audisp-syslog' | grep '{rule}' | grep 'AUID="test_rwuser"' """
+            result = duthost.shell(cmd)['stdout_lines']
+            assert len(result) > 0, f"Auditd {rule} rule does not contain the expected logs"
+
+
+def test_auditd_host_failure(localhost, duthosts, enum_rand_one_per_hwsku_hostname):
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+    container_name = "auditd_watchdog"
+    verify_container_running(duthost, container_name)
+
+    # Simulate auditd service failure
+    duthost.shell("sudo systemctl stop auditd")
+
+    output = duthost.command(DOCKER_EXEC_CMD.format(container_name) +
+                             "'{} {}'".format(NSENTER_CMD, CURL_HTTP_CODE_CMD), module_ignore_errors=True)["stdout"]
```

>module_ignore_errors=True [](http://example.com/codeflow?start=80&length=25)

While module_ignore_errors=True prevents test crashes, it also allows silent failures, potentially leading to false positives.

---

In this test case, it is intended to have failure, so `module_ignore_errors=True` is by design